src/index.ts

```diff
@@ -1,279 +1,122 @@
-		await perHeader(chain, api, header, extendedAccounts, reporters, methodSub)
-	});
-}
-
-async function listAllChains(config: AppConfig, reporters: Reporter[]) {
-	const _ = await Promise.all(config.endpoints.map((e) => listenChain(e, config.listen, config.accounts, config.method_subscription, reporters)));
-	// a rather wacky way to make sure this function never returns.
-	return new Promise(() => {});
-}
 
 async function main() {
-	if (!argv.c) {
-		logger.error('-c or DOT_NOTIF_CONF env variable must specify a config file');
-		process.exit(1);
-	}
-
-	const config: AppConfig = JSON.parse(readFileSync(argv.c).toString());
-	const missingField = missingAppConfig(config);
-	if (missingField) {
-		logger.error(`missing some key in config file: ${missingField}`);
-		process.exit(1);
-	}
-
-	if (config.listen !== ApiSubscription.Finalized && config.listen !== ApiSubscription.Head) {
-		logger.warn(`"listen" config not provided or invalid (${config.listen}), overwriting to ${ApiSubscription.Finalized}`);
-		config.listen = ApiSubscription.Finalized;
-	}
-
-	const reporters: Reporter[] = [];
-	for (const reporterType in config.reporters) {
-		if (reporterType === "email") {
-			const reporter = new EmailReporter(config.reporters[reporterType] as EmailConfig);
-			await reporter.verify();
-			reporters.push(reporter)
-		}
-		if (reporterType === "console") {
-			reporters.push(new ConsoleReporter())
-		}
-		if (reporterType === "fs") {
-			reporters.push(new FileSystemReporter(config.reporters[reporterType] as FsConfig))
-		}
-		if (reporterType === "matrix") {
-			const reporter = new MatrixReporter(config.reporters[reporterType] as MatrixConfig);
-			reporters.push(reporter)
-		}
-	}
-
-	if (config.accounts.length) {
-		config.accounts.forEach(([address, nick]) => logger.info(`📇 registering address ${address} aka ${nick}.`))
-	} else {
-		logger.info(`⚠️ no list of accounts provided, this will match with anything.`);
-	}
-	logger.info(`👂 method subscription: ${JSON.stringify(config.method_subscription)}`);
-
-	const retry = true;
-	while (retry) {
-		try {
-			await listAllChains(config, reporters);
-			// this is just to make sure we NEVER reach this code. The above function never returns,
-			// unless if we trap in an exception.
-			process.exit(1);
-		} catch (e) {
-			logger.error(`retrying due to error: ${e}`)
-		}
-
-	}
+    // Prepare readiness probe, defaults to false.
+    const readiness = new Healthprobe();
+    readiness.listen();
+
+    const configurator = new Config()
+    const config: AppConfig = configurator.config;
+    const reporters: Reporter[] = configurator.getReporters();
+
+    // This function is passed to chainMonitor.subscribeHandler and is called every block.
+    // Here is the main application logic that determines if a report is created
+    // and which extrinsics/events are included
+    const BlockHandler = async (blockHeader: Header, chainMonitor: ChainMonitor) => {
```

Comment:
Would make the main code much easier to read if you make this a function rather than a const.